server/sonar-server/src/main/java/org/sonar/server/activity/ActivityService.java

```diff
@@ -45,8 +46,15 @@ public void save(Activity activity) {
       .setAction(activity.getAction())
       .setMessage(activity.getMessage())
       .setData(KeyValueFormat.format(activity.getData()))
+      .setProfileKey((String) activity.getData().get("profileKey"))
```

Comment:
If my understanding is correct, you just need to update the following line : https://github.com/SonarSource/sonarqube/blob/feature/tb/user_updated_at_qp/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ActiveRuleChange.java#L97.
Replace 
```
activity.setData("profileKey", getKey().qProfile()); 
```
by 
```
activity. setProfileKey(getKey().qProfile());
```